pkg/apis/options/legacy_options.go

```diff
@@ -0,0 +1,104 @@
+package options
+
+import (
+	"fmt"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/oauth2-proxy/oauth2-proxy/pkg/logger"
+)
+
+type LegacyOptions struct {
+	// Legacy options related to upstream servers
+	LegacyUpstreams LegacyUpstreams `cfg:",squash"`
+
+	Options Options `cfg:",squash"`
+}
+
+func NewLegacyOptions() *LegacyOptions {
+	return &LegacyOptions{
+		LegacyUpstreams: LegacyUpstreams{
+			PassHostHeader:  true,
+			ProxyWebSockets: true,
+			FlushInterval:   time.Duration(1) * time.Second,
+		},
+
+		Options: *NewOptions(),
+	}
+}
+
+func (l *LegacyOptions) ToOptions() (*Options, error) {
+	upstreams, err := l.LegacyUpstreams.convert()
+	if err != nil {
+		return nil, fmt.Errorf("error converting upstreams: %v", err)
+	}
+	l.Options.UpstreamServers = upstreams
+
+	return &l.Options, nil
+}
+
+type LegacyUpstreams struct {
```

Comment:
Should we have a matching `FlagSet` function for these options and embed that in the monster `options.NewFlagSet` like we do here?

https://github.com/oauth2-proxy/oauth2-proxy/blob/e02f99eb581a77bd4b674ebdcb07c8264904ae24/pkg/apis/options/logging.go#L31

Reply:
Yep! That's a good idea, will help us work out what has an hasn't been migrated! Will sort out later